### PR TITLE
DAOS-14419 control: Display disabled ranks by default

### PR DESCRIPTION
**Pull request by knard38**
### Description

Always display the disabled targets and remove the old associated options.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]

**Review by mjmac — APPROVED** [review]
The code changes LGTM. I wonder if `Features: control` is enough to validate the test changes, though. @daltonbohning: Do the changed tests get auto-added now?

File: src/control/cmd/daos/pool.go [hunk 1]
```diff
@@ -224,9 +224,8 @@ type poolCmd struct {
 
 type poolQueryCmd struct {
 	poolBaseCmd
-	ShowEnabledRanks  bool `short:"e" long:"show-enabled" description:"Show engine unique identifiers (ranks) which are enabled"`
-	ShowDisabledRanks bool `short:"b" long:"show-disabled" description:"Show engine unique identifiers (ranks) which are disabled"`
-	HealthOnly        bool `short:"t" long:"health-only" description:"Only perform pool health related queries"`
+	ShowEnabledRanks bool `short:"e" long:"show-enabled" description:"Show engine unique identifiers (ranks) which are enabled"`
```
Comment (kjacque) [inline src/control/cmd/daos/pool.go:227]: Conceptually this feels strange to me. Is the default to _only_ show disabled ranks?

Comment (knard-intel) [reply src/control/cmd/daos/pool.go:227]: Yep, it was a request from @mjmac and @Michael-Hennecke to always have the disabled ranks as this information is really important for the admin.  From my side, I also agree on this as displayed this information can help the admin to realize that some nodes are unavailable, and thus some admin tasks are needed.

Comment (kjacque) [reply src/control/cmd/daos/pool.go:227]: OK, that makes sense. Thanks for the further context. :)

**Review by kjacque — APPROVED** [review]

**Comment by knard-intel** [comment]
@daltonbohning , could you tell me if the functional test part are OK for you ?
If not please, could you tell me if there is some other Features tags that I should add.

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Comment by daltonbohning** [comment]
> The code changes LGTM. I wonder if `Features: control` is enough to validate the test changes, though. @daltonbohning: Do the changed tests get auto-added now?

Changes to ftest/util map to `pr`, so don't necessarily cover affected tests. But changes to specific tests do include those specific tests. In this case, I think `Features: control` is probably sufficient since tests referencing `disabled_ranks` were updated.

**Comment by knard-intel** [comment]
@daos-stack/daos-gatekeeper please could you land this PR with the following message:
- Title: `DAOS-14419 control: Display disabled ranks by default`
- Body: 
`Always display the disabled targets and remove the old associated options.`

**Pull request merged**